File: barseek-profile-mservice/src/main/java/com/example/barseek_profile_mservice/kafka/KafkaConsumerService.java
package com.example.barseek_profile_mservice.kafka;

import com.example.barseek_profile_mservice.kafka.events.UserCreatedEvent;
import com.example.barseek_profile_mservice.model.entity.UserProfile;
import com.example.barseek_profile_mservice.service.UserProfileService;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.annotation.KafkaHandler;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@KafkaListener(topics = "user.events",groupId = "profile-service-group")
public class KafkaConsumerService {

    private final UserProfileService userProfileService;

    @KafkaHandler
    public void handleUserCreatedEvent(UserCreatedEvent event) {
        UserProfile profile = UserProfile.builder()
                .userId(event.getUserId())
                .email(event.getEmail())
                .build();
        userProfileService.createProfile(profile);
    }
}

File: barseek-rating-mservice/barseek-rating-mservice/src/main/kotlin/com/example/barseek_rating_mservice/kafka/KafkaConsumerService.kt
package com.example.barseek_rating_mservice.kafka

import com.example.barseek_rating_mservice.dto.BarReviewCreatedEvent
import com.example.barseek_rating_mservice.dto.DrinkReviewCreatedEvent
import com.example.barseek_rating_mservice.service.RatingService
import org.springframework.kafka.annotation.KafkaHandler
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service

@Service
@KafkaListener(topics = ["review.events"], groupId = "rating-service-group")
class KafkaConsumerService(private val ratingService: RatingService) {

    @KafkaHandler
    fun handleBarReviewCreatedEvent(event : BarReviewCreatedEvent) {
        ratingService.updateBarRating(event.barId,event.score)
    }

    @KafkaHandler
    fun handleDrinkReviewCreatedEvent(event : DrinkReviewCreatedEvent) {
        ratingService.updateDrinkRating(event.drinkId,event.score)
    }

}